plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java

```diff
@@ -129,6 +134,14 @@ public ListeningExecutorService provideListeningExecutor(BigQueryConfig config)
             return listeningDecorator(newFixedThreadPool(config.getMetadataParallelism(), daemonThreadsNamed("big-query-%s"))); // limit parallelism
         }
 
+        @Provides
+        @Singleton
+        @ForBigQuery
+        public ExecutorService provideExecutor(CatalogName catalogName)
+        {
+            return newCachedThreadPool(daemonThreadsNamed("bigquery-" + catalogName + "-%s"));
```

Comment:
We probably should have an upper bound on the number of threads here, but I wasn't sure if we can come up with a constant value or should it be configurable.

There's also another executor right above, but I didn't want to mix them up. The naming can be improved here for sure, but I also didn't have good ideas.